libs/src/sdk/scripts/generateBootstrapConfig.ts

```diff
@@ -0,0 +1,101 @@
+import path from 'path'
+
+type EnvironmentConfig = {
+  CLAIM_DISTRIBUTION_CONTRACT_ADDRESS: string
+  ETH_OWNER_WALLET: string
+  ETH_PROVIDER_URL: string
+  ETH_REGISTRY_ADDRESS: string
+  ETH_TOKEN_ADDRESS: string
+  IDENTITY_SERVICE_ENDPOINT: string
+  WORMHOLE_ADDRESS: string
+}
+
+const envConfigs: Record<'staging' | 'production', EnvironmentConfig> = {
+  production: {
+    CLAIM_DISTRIBUTION_CONTRACT_ADDRESS:
+      '0x683c19E621A0F107a291fdAB38f80179809d61B5',
+    ETH_OWNER_WALLET: '0xC7310a03e930DD659E15305ed7e1F5Df0F0426C5',
+    ETH_PROVIDER_URL: 'https://eth.audius.co',
+    ETH_REGISTRY_ADDRESS: '0xd976d3b4f4e22a238c1A736b6612D22f17b6f64C',
+    ETH_TOKEN_ADDRESS: '0x18aAA7115705e8be94bfFEBDE57Af9BFc265B998',
+    IDENTITY_SERVICE_ENDPOINT: 'https://identityservice.audius.co',
+    WORMHOLE_ADDRESS: '0x6E7a1F7339bbB62b23D44797b63e4258d283E095'
+  },
+  staging: {
+    CLAIM_DISTRIBUTION_CONTRACT_ADDRESS:
+      '0x74b89B916c97d50557E8F944F32662fE52Ce378d',
+    ETH_OWNER_WALLET: '',
+    ETH_PROVIDER_URL: 'https://eth.staging.audius.co',
+    ETH_REGISTRY_ADDRESS: '0xF27A9c44d7d5DDdA29bC1eeaD94718EeAC1775e3',
+    ETH_TOKEN_ADDRESS: '0x5375BE4c52fA29b26077B0F15ee5254D779676A6',
+    IDENTITY_SERVICE_ENDPOINT: 'https://identityservice.staging.audius.co',
+    WORMHOLE_ADDRESS: '0xf6f45e4d836da1d4ecd43bb1074620bfb0b7e0d7'
+  }
+}
+
+const generateBootstrapConfig = async (config: EnvironmentConfig) => {
+  const contracts = new EthContracts({
+    ethWeb3Manager: new EthWeb3Manager({
+      identityService: new IdentityService({
+        identityServiceEndpoint: config.IDENTITY_SERVICE_ENDPOINT
+      }),
+      web3Config: {
+        ownerWallet: config.ETH_OWNER_WALLET,
+        providers: [config.ETH_PROVIDER_URL],
+        tokenAddress: config.ETH_TOKEN_ADDRESS,
+        registryAddress: config.ETH_REGISTRY_ADDRESS,
+        claimDistributionContractAddress:
+          config.CLAIM_DISTRIBUTION_CONTRACT_ADDRESS,
+        wormholeContractAddress: config.WORMHOLE_ADDRESS
+      }
+    }),
+    tokenContractAddress: config.ETH_TOKEN_ADDRESS,
+    registryAddress: config.ETH_REGISTRY_ADDRESS,
+    claimDistributionContractAddress:
+      config.CLAIM_DISTRIBUTION_CONTRACT_ADDRESS,
+    wormholeContractAddress: config.WORMHOLE_ADDRESS
+  })
+
+  const discoveryNodes =
+    await contracts.ServiceProviderFactoryClient.getServiceProviderList(
+      'discovery-node'
+    )
+  if (!discoveryNodes || discoveryNodes.length === 0) {
+    throw Error('Services not found')
+  }
+  const minVersion = await contracts.getCurrentVersion('discovery-node')
+  return {
+    minVersion,
+    discoveryNodes: discoveryNodes.map((node) => node.endpoint)
+  }
+}
+
+const writeBootstrapConfig = async () => {
+  const production = await generateBootstrapConfig(envConfigs.production)
+  const staging = await generateBootstrapConfig(envConfigs.staging)
+  const development = {
+    minVersion: '0.0.0',
+    discoveryNodes: ['http://localhost:5000']
```

Comment:
should be audius-protocol-discovery-provider-1 maybe?

Reply:
for some reason that's not working with my local 🤔 but i agree this makes sense I think

Reply:
you need sshuttle (audius-compose connect). are you running that?

Reply:
I wasn't - thanks! But it's still not working 😢